p4-16/spec/P4-16-spec.mdk

```diff
@@ -2092,36 +2092,49 @@ else
 
 #### Using saturating arithmetic example
 
-We demonstrate the use of saturating arithmetic for preventing both
-underflow when decrementing `ttl`, and caping the value at maximum
+The most common use case for saturating arithmetic is staturating counters.
+Below is an example of using the PSA `Counter`extern as a saturating counter:
+
+~ Begin P4Example
+#ifdef HAVE_SATURATING
+typedef saturating_bit<16> ByteCounterT;
+#else
+typedef bit<16> ByteCounterT;
+#endif
+
+control C(in psa_ingress_input_metadata_t  istd) {
+  Counter<ByteCounterT, bit<10>> (1023, PSA_CounterType_t.BYTES) byte_counter;
```

Comment:
While it is totally OK to have an extern that counts something using saturating arithmetic, I do not see how useful it is to instantiate such an extern using a P4 type, since this counter can't be retrieved or anyhow accessed in the data plane and you would not even use the addition operation here. 

Are you saying that in PSA we should also instantiate counters using bit<n> to indicate the width of the underlying HW registers?

Reply:
Indeed, the way I would do this is actually

```
extern Counter {
   Counter(bit<32> width, bit<32> size, bool saturated);
}
```